chapter2.adoc

```diff
@@ -134,68 +132,147 @@ The privilege level of the trigger chain is determined by the trigger enabled fo
-| Register         |  Access condition
-| tselect(0x7a0)   |  mdbgen == 1 \|\| sdedbgalw == 1
-| tdata1(0x7a1)    |  mdbgen == 1 \|\| sdedbgalw == 1
-| tdata2(0x7a2)    |  mdbgen == 1 \|\| sdedbgalw == 1
-| tdata3(0x7a3)    |  mdbgen == 1 \|\| sdedbgalw == 1
-| tinfo(0x7a4)     |  mdbgen == 1 \|\| sdedbgalw == 1
-|================================================================
-
-The fields in mcontrol, mcontrol6, icount, itrigger, etrigger, and tmexttrigger are read/write accessible only when the access conditions are met. When access is disallowed, writes to these fields are ignored, and reads return zero.
+The following CSRs are provided for debugging supervisor domain. They are only read/write in Debug Mode and not accessible by hart.  
 
+.Allocated addresses for supervisor shadow of Debug Mode CSR  
 [options="header"]
-[cols="20%,80%"]
-.Tdata1 fields access condtion against privilege granted to external debugger
-|====================================
-| Field | Access condition
-| m     | mdbgen == 1                           
-| s     | mdbgen == 1 \|\| sdedbgalw == 1    
-| u     | mdbgen == 1 \|\| sdedbgalw == 1   
-| vs    | mdbgen == 1 \|\| sdedbgalw == 1  
-| vu    | mdbgen == 1 \|\| sdedbgalw == 1
-|====================================
-
-=== Other CSR updates
+[cols="25%,25%,50%"]
+|============================================================================================
+| Number   | Name       | Descirption
+| 0xaaa    | sdcsr      | Supervisor debug control and status register.
+| 0xaaa    | sdpc       | Supervisor debug program counter. 
+| 0xaaa    | sdcratch0  | Supervisor debug scratch register 0.
+| 0xaaa    | sdcratch1  | Supervisor debug scratch register 1.
+|============================================================================================
 
-==== Debug Control and Status (dcsr)
+The `sdcsr` provides supervisor access to `dcsr` and is a subset of it. The accessible fields are formatted as shown in <<sdcsr32>> and <<sdcsr64>>. When `mdbgen` is 0 and `sdedbgalw` is set to 1, the `prv` and `v` fields indicate the the privilege level at which the hart was executing before entering Debug Mode. Meanwhile, the `sdpc` is updated with the address of the next instruction to executed upon entry into Debug Mode. When resuming from Debug Mode, the privilege level of the hart is restored to the values in `prv` and `v`, while the hart's PC is updated wth the address in `sdpc`.   
 
-The dcsr is always accessible in Debug Mode. The access rules for field `prv` and `v` are addressed in subsection <<prvvacc>>. Beside `prv` and `v`, the access condition of remaining fields are listed in the following table.
-When the access conditions are met, they are read/write accessible. When access is disallowed, writes to these fields are ignored, and reads return zero.
+The `sdscratch0` and `sdscratch1` are optional scratch registers are only required if dscratch0 and dscratch1 are implemented.
```

Comment:
Arguably one could opt to implement more dscratch* regs for an M-mode debugger than an S-mode debugger, right?  Any reason there couldn't be dscratch[01] but only sdscratch0?

Reply:
I'm only half paying attention to this but I did see this comment just now.  Note that the DM has the hartinfo.nscratch field that says how many dscratch registers there are.  You might need another field in the DM if you want different numbers of scratch registers per mode.  Or just always have the same number for all modes.

Reply:
Ah okay, that's as good a reason as any to keep them consistent then.